tests/test_core.py

```diff
@@ -7822,6 +7826,18 @@ def test_dfe(self):
     self.emcc_args += ['-g2'] # test for issue #6331
     self.do_run_in_out_file_test('tests', 'core', 'test_hello_world')
 
+  def test_postrun_exception(self):
+    # verify that an exception thrown in postRun() will not trigger the
+    # compilation failed handler, and will be printed to stderr.
+    self.add_post_run('ThisFunctionDoesNotExist()')
+    src = open(path_from_root('tests', 'core', 'test_hello_world.c')).read()
+    self.build(src, self.get_dir(), 'src.c')
```

Comment:
Should we
A) also run this with sync instantiation?
B) skip this test if we have sync instantiation?

Actually, broader in scope than this review, should we have a binaryen-sync test suite, where we run everything with synchronous initialization?

Reply:
Speaking of sync, does this test pass in asmjs?

Reply:
Yup.